src/dodal/p45.py

```diff
@@ -0,0 +1,73 @@
+from ophyd import Component as Cpt
+from ophyd import EpicsMotor, MotorBundle
+from ophyd.areadetector.base import ADComponent as Cpt
+from pydantic import BaseSettings
+
+from dodal.devices.areadetector import AdAravisDetector
+
+
+# Settings can be customized via environment variables
+class Settings(BaseSettings):
+    pv_prefix: str = "BL45P"
+
+
+_settings = Settings()
+
+
+class SampleY(MotorBundle):
+    """
+    Motors for controlling the sample's y position and stretch in the y axis.
+    """
+
+    base: EpicsMotor = Cpt(EpicsMotor, "CS:Y")
+    stretch: EpicsMotor = Cpt(EpicsMotor, "CS:Y:STRETCH")
+    top: EpicsMotor = Cpt(EpicsMotor, "Y:TOP")
+    bottom: EpicsMotor = Cpt(EpicsMotor, "Y:BOT")
+
+
+class SampleTheta(MotorBundle):
+    """
+    Motors for controlling the sample's theta position and skew
+    """
+
+    base: EpicsMotor = Cpt(EpicsMotor, "THETA:POS")
+    skew: EpicsMotor = Cpt(EpicsMotor, "THETA:SKEW")
+    top: EpicsMotor = Cpt(EpicsMotor, "THETA:TOP")
+    bottom: EpicsMotor = Cpt(EpicsMotor, "THETA:BOT")
+
+
+class TomoStageWithStretchAndSkew(MotorBundle):
+    """
+    Grouping of motors for the P45 tomography stage
+    """
+
+    x: EpicsMotor = Cpt(EpicsMotor, "X")
+    y: SampleY = Cpt(SampleY, "")
+    theta: SampleTheta = Cpt(SampleTheta, "")
```

Comment:
Nit: Why not do these as ophyd v2 devices? I believe motors are basically done?

Reply:
Nit: Given this file is basically the interface to the user writing plans I was thinking it would be better to just have the `make_` functions here. It avoids a user accidentally writing:
```
from p45 import SampleY
my_stage = SampleY()
...
```
and wondering why everything is broken

Reply:
Makes sense, but then where should be put very-beamline-specific devices? `dodal/devices/<beamline>.py`?

Reply:
RE: Ophyd v2, probably a good idea to move to it sooner rather than later, but we want to prove the initial stack with Ophyd v1 first. @RAYemelyanova has a ticket to make some Ophyd v2 devices for P45 so those should go in here whenever they are ready (not urgent, Rose, just thought it was worth letting you know).

Reply:
I think `dodal/devices/<beamline>.py` is fine, yh